veps/sig-compute/hypervisor-abstraction.md

```diff
@@ -0,0 +1,577 @@
+- A scoped abstraction keeps the project approachable for contributors while unlocking new hardware backends.
+
+## Goals
+
+- Document the cluster-wide hypervisor configuration and per-component extension points (defaults, converter, webhooks, node labeller) so downstream implementations can extend behavior without invasive changes to existing components.
+- Resolve the hypervisor to be used for a VMI early and feed it through the virt-launcher converter so hypervisor-specific behavior stays localized.
+- Support both admission-time validation and mutation so administrators can enforce guardrails while still customizing VMIs for a given hypervisor.
+- Let VMs request hypervisor-specific allocatable device resources, avoiding new scheduling primitives.
+- Make it simple for downstreams to implement new hypervisors by following a documented contract.
+
+## Non Goals
+
+- Deliver a full implementation of any specific new hypervisor backend.
+- Redesign the VirtualMachineInstance API schema beyond additive fields.
+- Replace existing Hyper-V enlightenment features or other architecture-specific helpers.
+- Mandate new observability requirements; telemetry hooks remain optional.
+
+## Definition of Users
+
+- Cluster administrators who need to bootstrap KubeVirt on hardware that exposes alternative virtualization devices.
+- VM owners who would like to run a virtual machines using a non-KVM hypervisor with its differentiated capabilities
+- Platform vendors integrating proprietary or emerging hypervisor stacks with KubeVirt.
+- Upstream maintainers of core virt-launcher, virt-controller, and virt-handler.
+- Hypervisor-specific experts maintaining hypervisor-specific logic and validations.
+
+## User Stories
+
+1. As a cluster administrator, I can I would like to deploy KubeVirt on a cluster with non-KVM hypervisor nodes, and have non-KVM VMs schedule only on nodes that expose its required devices.
+2. As a platform engineer, I can supply hypervisor-specific VMI spec mutations and libvirt domain adjustments without forking the virt-launcher converter.
+3. As a core maintainer, I can maintain and develop the core of KubeVirt without deep knowledge of all specific hypervisors. 
+4. As a hypervisor-specific expert I know exactly where to add hypervisor-specific validation, testing and documentation when a new hypervisor is introduced, letting me develop quickly and independently.
+
+## Repos
+
+- [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt)
+- [kubevirt/enhancements](https://github.com/kubevirt/enhancements) (this VEP)
+
+## Design
+
+### Hypervisor Extension Points
+
+Cluster configuration (`spec.configuration.hypervisorConfiguration`) declares the list of supported hypervisors for the KubeVirt installation, and each control-plane package exposes focused extension contracts so downstream implementations only touch the areas they actually need:
```

Comment:
I wonder if that's really needed at this point.

If we indeed aim to support a single cluster-wide accelerator at this point, and we're not sure how the API will look like in the future, perhaps it's better to somehow inject this setting in a more manual way, e.g. through command line flags to the different components or a build-time argument.

Reply:
Thinking about it again, I agree that adding `spec.configuration.hypervisorConfiguration` while limiting it to a single hypervisor sounds reasonable.